script.go

```diff
@@ -967,3 +968,36 @@ func newScanner(r io.Reader) *bufio.Scanner {
 	scanner.Buffer(make([]byte, 4096), math.MaxInt)
 	return scanner
 }
+
+// EncodeBase64 produces the base64 encoding of the input.
+func (p *Pipe) EncodeBase64() *Pipe {
+	return p.Filter(func(r io.Reader, w io.Writer) error {
+		output := new(strings.Builder)
```

Comment:
Do we actually need to go via a string?

Reply:
@bitfield changed this to use a `bytes.Buffer` - does this work better?

Reply:
My point is really that `base64` has a byte-stream-based API (`Encoder`), and `Filter` is inherently a byte-stream-oriented function. Does it make sense to create an `Encoder` on `w` and `io.Copy` to it?

https://pkg.go.dev/encoding/base64#example-NewEncoder

Reply:
that's a very elegant solution! thanks for the suggestion